### PR TITLE
FW-4407 Video thumbnail generation

### PR DESCRIPTION
**Pull request by NolanVH**
### Description of Changes
These changes add thumbnail generation to the Video model.

The following three fields have been added to the model:
- `thumbnail` (containing a generated image with a max size of 100px)
- `small` (containing a generated image with a max size of 560px)
- `medium` (containing a generated image with a max size of 1000px)

The resized thumbnail images are generated using FFmpeg and saved the same way as thumbnails for the Image model. The [ffmpeg-python](https://github.com/kkroening/ffmpeg-python) bindings are used to generate the thumbnails, which requires that FFmpeg is already installed on the system. Install instructions can be found in the README.

FFmpeg installation has been added to the [Dockerfile](https://github.com/First-Peoples-Cultural-Council/fv-be/blob/7ba5a90142a5f1222dcaa16aa5f96aee26e8c137/Dockerfile#L10), as well as the [Run Pytest workflow](https://github.com/First-Peoples-Cultural-Council/fv-be/blob/7ba5a90142a5f1222dcaa16aa5f96aee26e8c137/.github/workflows/run-pytest.yml#L41-L42).

### Checklist
- [x] README / documentation has been updated if needed
- [x] PR title / commit messages contain Jira ticket number if applicable
- [x] Tests have been updated / created if applicable
- [x] Admin Panel has been updated if models have been added or modified
- [x] Migrations have been updated and committed if applicable

### Reviewers Should Do The Following:
- [x] Review the code changes here
- [x] Pull the branch and test locally

### Additional Notes
N/A

### CODE REVIEW

File: firstvoices/backend/models/media.py [hunk 1]
```diff
@@ -208,51 +245,6 @@ def generate_resized_images(self):
...
-            self.thumbnail.delete(save=False)
-            self.small.delete(save=False)
-            self.medium.delete(save=False)
-        except Exception as e:
-            # this will only happen for connection or permission errors, so it's a warning
-            self.logger.warn(
-                f"Failed to delete file from S3 when deleting [{str(self)}]. Error: {e} "
-            )
-
-        return result
```
Comment (NolanVH) [inline firstvoices/backend/models/media.py:254]: Moved these to the `MediaBase` class as they are needed for both the `Image` and `Video` models.

File: README.md [hunk 2]
```diff
@@ -5,6 +5,7 @@ Backend for the FirstVoices application
 [![Built with Cookiecutter Django](https://img.shields.io/badge/built%20with-Cookiecutter%20Django-ff69b4.svg?logo=cookiecutter)](https://github.com/cookiecutter/cookiecutter-django/)
```
[the remark this answers is not in the document]
Comment (sarahfirstvoices) [reply README.md:5]: I don't see much benefit to the codestyle badget either, unless someone loves it for some reason

File: README.md [hunk 3]
```diff
@@ -5,6 +5,7 @@ Backend for the FirstVoices application
...
 [![Black code style](https://img.shields.io/badge/code%20style-black-000000.svg)](https://github.com/ambv/black)
 [![Quality Gate Status](https://sonarcloud.io/api/project_badges/measure?project=First-Peoples-Cultural-Council_fv-be&metric=alert_status)](https://sonarcloud.io/summary/new_code?id=First-Peoples-Cultural-Council_fv-be)
+[![Run Pytest](https://github.com/First-Peoples-Cultural-Council/fv-be/actions/workflows/run-pytest.yml/badge.svg?branch=main)](https://github.com/First-Peoples-Cultural-Council/fv-be/actions/workflows/run-pytest.yml)
```
Comment (sarahfirstvoices) [inline README.md:8]: I think we could make the text of this just "Tests"

**Review by sarahfirstvoices — CHANGES_REQUESTED** [review]

**Comment by sarahfirstvoices** [comment]
I merged some model changes that are going to have conflicts for you, but might make it easier to reuse some of the thumbnail code. Call me if you want to talk it through when you get back!

File: README.md (hunk 3 above)
Comment (NolanVH) [reply README.md:8]: It looks like the only way to change this is to update the name of the workflow, which is quick and easy but do we want to change the name there as well? Updating the text in the README does not change the image that is generated by GitHub.

Comment (sarahfirstvoices) [reply README.md:6]: Oh gotcha. This is fine then

File: firstvoices/backend/models/media.py [hunk 4]
```diff
@@ -58,6 +61,12 @@ def __str__(self):
         return f"{self.content.name} ({self.site})"
 
     def save(self, **kwargs):
+        if not self._state.adding:
+            raise NotSupportedError(
+                "Editing existing files is not supported at this time. Please create a new file if you would like to "
+                "update a media file."
+            )
```
Comment (sarahfirstvoices) [inline firstvoices/backend/models/media.py:68]: Thanks for adding this.

**Review by sarahfirstvoices — APPROVED** [review]

**Pull request merged**